### ISSUE TITLE
Adventure map, AI turn, disappearance of wandering creatures following an attack by an AI hero is not shown

### ISSUE THREAD

**Issue by LeHerosInconnu** [ihhub/fheroes2#8001]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

In fheroes2, during the AI's turn, the disappearance of wandering creatures following an attack by an AI hero is not shown.

In fheroes2:

![Before AI hero attacks orcs 01](https://github.com/ihhub/fheroes2/assets/43583994/2000a0f0-05c0-4aba-b978-e3adf4904f42)

https://github.com/ihhub/fheroes2/assets/43583994/d1edef6b-858a-4ce6-a3ea-dfa27a60c6f3

Here is the save file:
[Before AI hero attacks orcs 0.zip](https://github.com/ihhub/fheroes2/files/13198177/Before.AI.hero.attacks.orcs.0.zip)

### Describe the possible solution

The player should be given time to see that the creatures have disappeared from the adventure map before continuing.
The same should apply to other objects such as resource stacks, artifacts, etc.

**Comment by ihhub** [comment]
Hi @LeHerosInconnu , is it the same in the original game?

**Comment by LeHerosInconnu** [comment]
Hello @ihhub,

> Hi @LeHerosInconnu , is it the same in the original game?

I didn't check in the original game for now.
Anyway, it would be an improvement on the original game.

**Comment by oleg-derevenetz** [comment]
I'm afraid that such an innovation will slow down the AI turns, which will only annoy.

**Comment by LeHerosInconnu** [comment]
Hello @ihhub and @oleg-derevenetz,

> I'm afraid that such an innovation will slow down the AI turns, which will only annoy.

It doesn't have to take an exaggerated amount of time, just long enough for the player to see clearly that the AI hero has picked up a pile of resources or has fought a troop of wandering creatures.
This will always be better than having to find an opposing AI hero on the adventure map, during the human player's turn, to check whether the AI hero has picked up a pile of resources or fought a troop of wandering creatures.
Which, by the way, is really annoying.

Also, disappearances/actions are already shown for AI heroes, but not for all captures/actions, why this state of facts?

**Comment by oleg-derevenetz** [comment]
> Also, disappearances/actions are already shown for AI heroes

And they also annoy, especially fade-in/fade-out of teleporting AI heroes or AI heroes who use Whirlpools. If this would be observed for each resource pile or monster pack as well, then this would be even more annoying. Battles between heroes (when heroes are disappear with fade-out) are relatively rare in comparison with resource collection or elimination of wandering monsters.

**Comment by LeHerosInconnu** [comment]
@ihhub and @oleg-derevenetz.

> Hi @LeHerosInconnu , is it the same in the original game?

I checked in the original game and all the actions of collecting resources, capturing an artifact, fighting a troop of wandering creatures (with victory or defeat for the AI hero), etc., are clearly shown to the human player during the AI's turn.

> > Also, disappearances/actions are already shown for AI heroes
> 
> And they also annoy, especially fade-in/fade-out of teleporting AI heroes or AI heroes who use Whirlpools. If this would be observed for each resource pile or monster pack as well, then this would be even more annoying. Battles between heroes (when heroes are disappear with fade-out) are relatively rare in comparison with resource collection or elimination of wandering monsters.

If this annoys the user, he can always set the "Ennemy Speed" to "Jump" or "Don't Show".

**Comment by oleg-derevenetz** [comment]
> I checked in the original game and all the actions of collecting resources, capturing an artifact, fighting a troop of wandering creatures (with victory or defeat for the AI hero), etc., are clearly shown to the human player during the AI's turn.

As far as I can see in the original game there is also no special animation during the AI turn when AI-controlled hero is collecting resources:

https://github.com/ihhub/fheroes2/assets/32623900/ca0c566f-9b19-455e-9b37-2a3d1bcd4aa1

They disappear instantly and hero continues its turn.

> If this annoys the user, he can always set the "Ennemy Speed" to "Jump" or "Don't Show".

"Jump" will not help here because it affects only the hero's movement speed and not the speed of animation of disappearing objects. Perhaps we should do something with this in general - e.g. scale the AI-related animations depending on the AI turn speed setting.

**Comment by LeHerosInconnu** [comment]
@ihhub and @oleg-derevenetz.

> > I checked in the original game and all the actions of collecting resources, capturing an artifact, fighting a troop of wandering creatures (with victory or defeat for the AI hero), etc., are clearly shown to the human player during the AI's turn.
> 
> As far as I can see in the original game there is also no special animation during the AI turn when AI-controlled hero is collecting resources:
> 
> https://github.com/ihhub/fheroes2/assets/32623900/ca0c566f-9b19-455e-9b37-2a3d1bcd4aa1
> 
> They disappear instantly and hero continues its turn.

My point is not to show the disappearing animation of resources and the like, but rather the fact that the resource is removed from the adventure map (instantly, as in the original game) so that the player can see it clearly on the adventure map during the AI's turn.
In the video included as an example of the original game, you can clearly see all the actions of capturing resource stacks, treasure chests, artifacts, etc. performed by the AI heroes.
Fheroes2 should also show this.

In the video of the original post, we don't see that the troop of wandering creatures has been removed from the adventure map following the attack of an AI hero, this is not shown.
Animation of the disappearance of the troop of wandering creatures is not necessary, it's just necessary to show the human player that the troop of wandering creatures is no longer on the adventure map, having been defeated by the AI hero.
In the original game, the troops of wandering creatures defeated by an AI hero also instantly disappear from the adventure map (without animation), this is clairly shown to the human player during the AI's turn.

> > If this annoys the user, he can always set the "Ennemy Speed" to "Jump" or "Don't Show".
> 
> "Jump" will not help here because it affects only the hero's movement speed and not the speed of animation of disappearing objects. Perhaps we should do something with this in general - e.g. scale the AI-related animations depending on the AI turn speed setting.

It's a good idea to set the speed of animations with the AI speed setting.

**Comment by oleg-derevenetz** [comment]
> In the video included as an example of the original game, you can clearly see all the actions of capturing resource stacks, treasure chests, artifacts, etc. performed by the AI heroes.
Fheroes2 should also show this.

I think that in the original game, in the case when a hero eliminates a stack of monsters, after which he has no movement points left and the camera switches to another hero (or maybe even another AI player), we will also not have time to see the disappearance of a group of monsters. In my video, they are visible only because the hero is still moving for a while after that.

**Comment by LeHerosInconnu** [comment]
@ihhub and @oleg-derevenetz.

> > In the video included as an example of the original game, you can clearly see all the actions of capturing resource stacks, treasure chests, artifacts, etc. performed by the AI heroes.
> > Fheroes2 should also show this.
> 
> I think that in the original game, in the case when a hero eliminates a stack of monsters, after which he has no movement points left and the camera switches to another hero (or maybe even another AI player), we will also not have time to see the disappearance of a group of monsters. In my video, they are visible only because the hero is still moving for a while after that.

We can then do it better and show it in all cases in fheroes2.

**Comment by oleg-derevenetz** [comment]
> We can then do it better and show it in all cases in fheroes2.

Yes, but first we need to implement some adjustable delay depending on the AI speed setting, otherwise the players will curse us.

**Comment by LeHerosInconnu** [comment]
@ihhub and @oleg-derevenetz.

> > We can then do it better and show it in all cases in fheroes2.
> 
> Yes, but first we need to implement some adjustable delay depending on the AI speed setting, otherwise the players will curse us.

As a well known commercial slogan says:

“Just Do It.” :)

**Comment by Branikolog** [comment]
Hello everyone.
I heard multiple complains from players regarding fade animations of AI heroes during AI turn. I mean slow rendering speed of this animations. Lots of people would love to see speed up those fade animations for AI heroes depending on AI hero speed selected. If this would be implemented we can use such speeded fade animations for resources to make picking by AI heroes more noticeable, but not slowing AI turn too much.
If the implementation is not really trivial, can we just add a pause before and after AI hero picked resource? Like adding a few static frames before and after AI hero picked a resource, so a human player will notice this action more easily? Current implementation implies such logic, when different AI heroes could make actions in random order. And sometimes this could be a single action, when hero picks a resource pile and then game refocuses on other hero in a moment. So engine focuses on such hero just for 1 frame and the action occurred could be barely noticeable.
